### PR TITLE
notebooks in the browser options

### PR DESCRIPTION
**Pull request by minsa110**
For https://github.com/microsoft/vscode-jupyter-internal/issues/381

### CODE REVIEW

File: docs/datascience/nb-in-the-browser.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+TOCTitle: Jupyter Notebooks on VS Code browser
+ContentId: 0faf5b06-ddad-4594-8d5e-fa409c7da82c
+PageTitle: Jupyter Notebooks on VS Code browser
+DateApproved: 1/9/2023
+MetaDescription: Ways to work with Jupyter notebooks in the browser on VS Code.
+MetaSocialImage: images/tutorial/social.png
+---
+
+# Jupyter notebooks in VS Code running in the browser
+
```
Comment (gregvanl) [inline docs/datascience/nb-in-the-browser.md:13]: Good to add a couple sentence introduction here before jumping into specifics.
Imagine someone landing here via search and not necessarily knowing what a Jupyter notebook is or is used for or that there are VS Code for the Web or Codespaces options along with VS Code desktop

**Comment by gregvanl** [comment]
Can we change the filename to notebooks-web.md? That URL should do better with SEO and also matches our vscode-web.md and python-web.md naming.
We should also loop in @bamurtaugh to help with the naming of the topic/feature. I think we like to say "for the Web" or "in the Web" rather than "in the browser".

**Comment by bamurtaugh** [comment]
Thanks for looping me in, @gregvanl! Let's stick with "for the Web" as possible, since that's consistent with the official "VS Code for the Web" naming.

If it feels awkward at any point (i.e. like too formal of a title for what you're describing), I think lowercase "for the web" or even "in the web" would be alright.

File: docs/datascience/nb-in-the-browser.md (hunk 1 above)
Comment (minsa110) [reply docs/datascience/nb-in-the-browser.md:13]: Added intro!

**Comment by minsa110** [comment]
Thanks @gregvanl and @bamurtaugh ! The document talks about all browser options, including Codespaces. Does "for the web" apply to Codespaces too? If so, I'm happy to rename it "Jupyter notebooks on VS Code for the Web"

Will rename the doc `notebooks-web.md`

**Comment by minsa110** [comment]
> Thanks @gregvanl and @bamurtaugh ! The document talks about all browser options, including Codespaces. Does "for the web" apply to Codespaces too? If so, I'm happy to rename it "Jupyter notebooks on VS Code for the Web"

Even if it doesn't, I agree we should change the title. The current one feels wordy and awkward :/

**Comment by gregvanl** [comment]
Given the various technologies in the article, I'm leaning towards "Jupyter Notebooks on the web" for the TOCTitle, PageTitle, and article H1.

File: docs/datascience/nb-in-the-browser.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+3. That's it! A codespace will be created for you to get started!
+   > **Tip**: You can customize your project by committing [configuration files](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers) to your repository (often known as Configuration-as-Code), which creates a repeatable codespace configuration for all users of your project. You can also see examples of predefined configurations on the [devcontainers/images repository](https://github.com/devcontainers/images/tree/main/src) to use and/or amend as needed for your project.
+
+## Jupyter notebooks on VS Code web (vscode.dev)
+
+You can also use [VS Code for the web](https://code.visualstudio.com/docs/editor/vscode-web) by going to https://vscode.dev or https://github.dev (available by typing '.' when viewing a repo on https://github.com ), which gives you options to connect to the following types of kernels for your notebook:
+1. [Remote tunneling with VS Code Server](#remote-tunneling-with-vs-code-server)
+2. [Remote tunneling with Jupyter server](#connect-to-a-remote-jupyter-server)
+
+### Remote tunneling with VS Code Server
```
Comment (gregvanl) [inline docs/datascience/nb-in-the-browser.md]: Can you connect to a server via a remote tunnel from the VS Code Desktop as well?

File: docs/datascience/nb-in-the-browser.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+1. Install the `code` [CLI](https://code.visualstudio.com/#alt-downloads) on your remote machine.
+   > **Note**: You can skip this step if VS Code Desktop is already installed on the remote machine, because the `code` CLI is built into VS Code Desktop.
+2. Create a [secure tunnel](https://code.visualstudio.com/docs/remote/tunnels) with the tunnel command: `code tunnel` (you can also run `code-insiders tunnel` if you wish to use the daily release of VS Code) and follow the prompts to grant access to the server. This will download and start the VS Code Server on your remote machine and then create a tunnel to it.
+3. Use the CLI's output vscode.dev URL tied to this remote machine, such as https://vscode.dev/tunnel/<machine_name>/<folder_name> for access to this remote machine on any client.
+4. Open a notebook file and select any Jupyter kernel or Python environment to run your code
+
+### Connect to a remote Jupyter server
+
+You can also connect to a remote Jupyter server directly by selecting the `Jupyter: Specify Jupyter Server for Connections` command -> pasting the URL for your remote Jupyter server, e.g., `http://<ip-address>:<port>/?token=<token>`.
```
Comment (gregvanl) [inline docs/datascience/nb-in-the-browser.md]: Use bold for command text

**Comment by gregvanl** [comment]
A couple of things for the images:
- Can you enable Git LFS in your local repo? This lets us keep large binary files out of the repo. Here's the incantation for enabling Git LFS https://github.com/microsoft/vscode-docs/blob/main/README.md#cloning Let me know if you have questions. You only need to set this up once.
- Can you also move the images to a 'notebook-web' folder to match the new filename?
Thanks!

File: docs/datascience/nb-in-the-browser.md (hunk 2 above)
Comment (minsa110) [reply docs/datascience/nb-in-the-browser.md]: Added a mention for it

File: docs/datascience/nb-in-the-browser.md (hunk 3 above)
Comment (minsa110) [reply docs/datascience/nb-in-the-browser.md]: Did it again 🤦🏻‍♀️ Done!

**Comment by minsa110** [comment]
> A couple of things for the images:
> 
> * Can you enable Git LFS in your local repo? This lets us keep large binary files out of the repo. Here's the incantation for enabling Git LFS https://github.com/microsoft/vscode-docs/blob/main/README.md#cloning Let me know if you have questions. You only need to set this up once.

Are you suggesting we add this info on this page? How about mentioning it in the intro for the "Jupyter notebooks on VS Code for the Web" section? I'm hesitant to give this too much of a spotlight because in many DS/ML cases, I see that users run out of LFS storage quota very quickly and it becomes unusable.


> * Can you also move the images to a 'notebook-web' folder to match the new filename?
>   Thanks!

Done :)

**Comment by gregvanl** [comment]
Sorry for the confusion, when people add or modify images or .gifs in the vscode-docs repo, we ask them to use Git LFS so that the images/.gifs are stored there not in the main repo.
You just need to enable Git LFS in your local repo (install Git LFS then run `git lfs install` once in the repo root folder). 
You might need to remove the images and commit them again so they are pushed to Git LFS and have the right Git LFS pointers.
I didn't mean we should mention Git LFS in the new article.

**Review by gregvanl — APPROVED** [review]
Thanks!

**Pull request merged**